packages/lab/src/badge/Badge.css

```diff
@@ -1,67 +1,40 @@
-/* Default variables applied to the root element */
-.saltBadge {
-  /* Colors */
-  --badge-content-text-color: var(--saltBadge-content-text-color, var(--salt-accent-foreground));
-  --badge-content-icon-color: var(--saltBadge-content-icon-color, var(--salt-accent-background));
-  /* Font, Typography */
-  --badge-content-borderRadius: 16px;
-  --badge-content-lineHeight: 15px;
-  --badge-content-padding: 0px 4px;
-  --badge-fontSize: var(--saltBadge-fontSize, var(--salt-text-label-fontSize));
-
-  /* Derived */
-  /* Same as content padding in x */
-  --badge-offset-inline: -4px;
-
-  --badge-sequence-gap: var(--salt-size-unit);
-}
-
 /* Style applied to the root element */
 .saltBadge {
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
   display: inline-flex;
-  position: relative;
   flex-shrink: 0;
-  vertical-align: middle;
-  line-height: var(--salt-text-lineHeight);
-  font-size: var(--salt-text-fontSize);
-}
-
-/* Give automatic margin when they sit next to each other */
-.saltBadge + .saltBadge {
-  margin-left: var(--badge-sequence-gap);
+  justify-content: center;
 }
 
-/* Style applied to the badge in the corner of the node */
 .saltBadge-badge {
-  color: var(--badge-content-text-color);
+  /* Should this vary according to touch size */
+  padding: 0px var(--salt-spacing-50);
+  line-height: var(--salt-text-lineHeight);
+  min-width: 10px;
+  max-width: 50px;
+  border-radius: 999px;
+
+  align-items: center;
+  justify-content: center;
   display: inline-block;
-  font-weight: 700; /* TODO: Use characteristic once aligned */
-  font-size: var(--badge-fontSize);
-  /* default small type */
-  width: auto;
-  min-width: 15px;
-  line-height: var(--badge-content-lineHeight);
-  border-radius: var(--badge-content-borderRadius);
-  text-align: center;
+  margin-bottom: 4px;
 
-  transform: translate(var(--badge-offset-inline), -50%);
-  transform-origin: 100% 0%;
-  /*
-    We want offset the next inline element to be strictly next to the badge.
-    This negative margin makes sure this node "appears" to be narrower.
-   */
-  margin-right: var(--badge-offset-inline);
+  font-size: var(--salt-text-label-fontSize);
+  font-weight: var(--salt-accent-fontSize);
+  background: var(--salt-accent-background);
+  color: var(--salt-accent-foreground);
 
-  padding: var(--badge-content-padding);
-  background: var(--badge-content-icon-color);
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
 
+.saltBadge-topRight {
   position: relative;
```

Comment:
@ZarkoPernar reverted to this positioning method. When using a wrapper for the child and badge it was difficult to keep the alignment of the badge consistent across Icons as well as text.